blocks/components/editable/index.js

```diff
@@ -271,14 +262,26 @@ export default class Editable extends wp.element.Component {
 		return !! this.state.formats[ format ];
 	}
 
-	toggleFormat( format ) {
+	changeFormats( formats ) {
 		this.editor.focus();
 
-		if ( this.isFormatActive( format ) ) {
-			this.editor.formatter.remove( format );
-		} else {
-			this.editor.formatter.apply( format );
-		}
+		Object.keys( formats ).forEach( format => {
+			const formatValue = formats[ format ];
+			if ( format === 'link' ) {
```

Comment:
Based on the distinctly unique treatment between links and other formatting, I'm wondering if they ought to be handled separately. Maybe not; I do like the single `onChange` prop applied to the toolbar component.